src/time_integration/paired_explicit_runge_kutta/methods_PERK4.jl

```diff
@@ -0,0 +1,326 @@
+
+    @assert isfile(path_a_coeffs) "Couldn't find file $path_a_coeffs"
+    a_coeffs = readdlm(path_a_coeffs, Float64)
+    num_a_coeffs = size(a_coeffs, 1)
+
+    @assert num_a_coeffs == num_coeffs_max
+    if num_coeffs_max > 0
+        a_matrix[1, :] -= a_coeffs
+        a_matrix[2, :] = a_coeffs
+    end
+
+    # Constant/non-optimized part of the Butcher matrix
+    a_matrix_constant = [(0.479274057836310-(0.114851811257441 / cS3)) 0.1397682537005989 0.1830127018922191
+                         0.114851811257441/cS3 0.648906880894214 0.028312163512968]
+
+    return a_matrix, a_matrix_constant, c
+end
+
+@doc raw"""
+    PairedExplicitRK4(num_stages, base_path_a_coeffs::AbstractString, dt_opt = nothing;
+                      cS3 = 1.0f0)
+    PairedExplicitRK4(num_stages, tspan, semi::AbstractSemidiscretization;
+                      verbose = false, cS3 = 1.0f0)
+    PairedExplicitRK4(num_stages, tspan, eig_vals::Vector{ComplexF64};
+                      verbose = false, cS3 = 1.0f0)
+
+    Parameters:
+    - `num_stages` (`Int`): Number of stages in the paired explicit Runge-Kutta (P-ERK) method.
+    - `base_path_a_coeffs` (`AbstractString`): Path to a file containing some coefficients in the A-matrix in 
+      the Butcher tableau of the Runge Kutta method.
+      The matrix should be stored in a text file at `joinpath(base_path_a_coeffs, "a_$(num_stages).txt")` and separated by line breaks.
+    - `dt_opt` (`Float64`, optional): Optimal time step size for the simulation setup. Can be `nothing` if it is unknown. 
+       In this case the optimal CFL number cannot be computed and the [`StepsizeCallback`](@ref) cannot be used.
+    - `tspan`: Time span of the simulation.
+    - `semi` (`AbstractSemidiscretization`): Semidiscretization setup.
+    - `eig_vals` (`Vector{ComplexF64}`): Eigenvalues of the Jacobian of the right-hand side (rhs) of the ODEProblem after the
+      equation has been semidiscretized.
+    - `cS3` (`Float64`, optional): Value of $c_{S-3}$ in the Butcher tableau, where
+      $S$ is the number of stages. Default is `1.0f0`.
+
+The following structures and methods provide an implementation of
+the fourth-order paired explicit Runge-Kutta (P-ERK) method
+optimized for a certain simulation setup (PDE, IC & BC, Riemann Solver, DG Solver).
+The method has been proposed in 
+- D. Doehring, L. Christmann, M. Schlottke-Lakemper, G. J. Gassner and M. Torrilhon (2024).
+  Fourth-Order Paired-Explicit Runge-Kutta Methods
+  [DOI:10.48550/arXiv.2408.05470](https://doi.org/10.48550/arXiv.2408.05470)
+"""
+struct PairedExplicitRK4 <: AbstractPairedExplicitRKSingle
+    num_stages::Int # S
+
+    # Optimized coefficients, i.e., flexible part of the Butcher array matrix A.
+    a_matrix::Union{Matrix{Float64}, Nothing}
+    # This part of the Butcher array matrix A is constant for all PERK methods, i.e., 
+    # regardless of the optimized coefficients.
+    a_matrix_constant::Matrix{Float64}
```

Comment:
If I understand it correctly, `a_matrix_constant` is the same for every constructor. Would it be a better idea to make it a global constant so we don't have to write it again and preserve the sense that it isn't computed differently for different kinds of constructors?